replica/table.cc

```diff
@@ -3515,11 +3530,20 @@ future<> table::clear_inactive_reads_for_tablet(database& db, storage_group& sg)
     }
 }
 
-future<> table::stop_compaction_groups(storage_group& sg) {
+future<> storage_group::stop() {
+    if (_async_gate.is_closed()) {
+        co_return;
+    }
     // Synchronizes with in-flight writes if any, and also takes care of flushing if needed.
-    for (auto& cg_ptr : sg.compaction_groups()) {
+    for (auto& cg_ptr : compaction_groups()) {
         co_await cg_ptr->stop();
     }
+    // carefully closed after stopping individual groups, such that cleanup won't be blocked on ongoing compaction.
+    co_await _async_gate.close();
+}
```

Comment:
@bhalevy there's something I realized I don't handle yet. what if tablet is migrated out during major? 

let's say tablet X is migrated out while major is running on it (with storage_group::async_gate taken). but given that we're closing gate after stopping groups (because we don't want tablet cleanup to wait on compaction), it can happen major fail due to it being stopped on compaction_group::stop(), which results in table wide op failing entirely.

Reply:
I don't think it's a problem since compaction_stopped_exception (triggered by compaction_group::stop()) is swallowed by compaction manager (used in shutdown path).

Reply:
right.  regular compaction could also run when a tablet is migrated and we should stop it too.